test/helpers.lua

```diff
@@ -741,17 +741,46 @@ function module.read_file_list(filename, start)
   if not file then
     return nil
   end
+
+  local chars_per_line = 256
```

Comment:
Instead of guessing, you could read the last N characters of the file and check if a newline was found among them; if it isn't found, then increment the tail offset and try again.

Can be a C function if Lua isn't able to do that natively (see e.g. https://stackoverflow.com/a/13790698).